cpp/src/transform/nats_to_nulls.cu

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */ 
+
+#include <bitmask/valid_if.cuh>
+
+namespace cudf {
+
+namespace detail {
+
+template <typename T>
+struct predicate_not_nat{
+
+  CUDA_HOST_DEVICE_CALLABLE
+  bool operator()(gdf_index_type index) const {
+      return !(static_cast<T*>(input.data)[index] == LLONG_MIN);
```

Comment:
`NaT` should be defined in a central location, like `types.hpp`.
```suggestion
      static constexpr NaT{ std::numeric_limits<cudf::timestamp>::lowest() };
      return !(static_cast<T*>(input.data)[index] == cudf::NaT);
```

Reply:
How about creating two NaTs, one for `timestamp` and `date64`. Or `std::numeric_limits<gdf_type>::lowest()` using directly in the predicate and handle the type over there.

Reply:
Two `NaT`s is fine. It should be:

```
cudf::timestamp::NaT
cudf::date64::NaT
```